modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -323,22 +313,25 @@ public void onException(ConnectionException exception) {
                        sourceAdapter.getName(), getLocation().getRootContainerName()),
                 exception);
 
-    Mono<Void> reconnectionAction = sourceAdapter.getReconnectionAction(exception)
-        .map(p -> from(retryPolicyTemplate.applyPolicy(p, retryScheduler)))
-        .orElseGet(() -> create(sink -> {
-          try {
-            exception.getConnection().ifPresent(sourceConnectionManager::invalidate);
-            restart(new ReactiveReconnectionCallback(sink));
-          } catch (Exception e) {
-            sink.error(e);
-          }
-        }));
-
-    reconnectionAction
-        .doOnSuccess(v -> onReconnectionSuccessful())
-        .doOnError(this::onReconnectionFailed)
-        .doAfterTerminate(() -> reconnecting.set(false))
-        .subscribe();
+    Executors.newSingleThreadExecutor().submit(() -> {
```

Comment:
bad usage of scheduler:

* You're creating a scheduler which you never dispose
* All scheduling should be done through the scheduling service